msu/classes/weighted_container.nut

```diff
@@ -79,6 +79,39 @@
 		}
 	}
 
+	// We expect 1 or 2 args.
+	// - When 1 it is an array
+	// - When 2 then the first arg is the weight and the second is the array
+	function addMany( ... )
```

Comment:
Why does this use vargv and not just array as first parameter, and weight with a default value of 1.0 as the second parameter?

Reply:
Because that is very bad for readability. When you have to pass a long array, and then a small weight at the end, it is very easy to miss that second parameter. Example:
```squirrel
local container = ::MSU.Class.WeightedContainer().addMany([
   something,
   something else,
   some morethings,
   even more stuff, 
   stuff,
   some other stuff,
   another stuff,
   some stuff,
   yes,
   more, 
   stuff],
   3);
```
vs
```squirrel
local container = ::MSU.Class.WeightedContainer().addMany(3, [
   something,
   something else,
   some morethings,
   even more stuff, 
   stuff,
   some other stuff,
   another stuff,
   some stuff,
   yes,
   more, 
   stuff]
```

Reply:
For me it's quite readable both ways, you just need to know if you look at the beginning or the end. It also conceptually makes more sense to me to have the stuff to add come first, and the weight come later, so it helps usability. I also quite dislike common sense values you have to pass just because, such as a default 1 in this case. Plus, it makes the function less fugly.

Reply:
You don't have to pass `1` in the case of vargv. 1 is the default if you just pass an array.

Reply:
I don't really mind either way, (either weight first and require or weight second and optional) but I'm actually not a huge fan of the vargv approach either